repos/system_upgrade/common/actors/scangrubdevice/actor.py

```diff
@@ -19,8 +19,8 @@ def process(self):
         if architecture.matches_architecture(architecture.ARCH_S390X):
             return
 
-        device_name = grub.get_grub_device()
-        if device_name:
-            self.produce(GrubInfo(orig_device_name=device_name))
+        devices = grub.get_grub_devices()
```

Comment:
IIUC orig_devices is an empty list by default, so L22-L26 should be equivalent to
```
self.produce(GrubInfo(orig_devices=devices))
```